### ISSUE TITLE
Include dotnet sdk 2.0.5 assemblies in the runtime package store

### ISSUE THREAD

**Issue by sartan** [cloudfoundry/dotnet-core-buildpack#177]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?

```
api_version: 2.98.0 (pcfbeta) and 2.101.0 (pws)
cf version: 6.34.0+f53d03a5e.2018-01-09
```

What version of the buildpack you are using?
```
2.0.0
```

If you were attempting to accomplish a task, what was it you were attempting to do?
Deploy an mvc app scaffolded with `dotnet` cli:

```
dotnet new mvc
dotnet publish --configuration Release
cf push testapp --random-route -p ./bin/release/netcoreapp2.0/publish/
```

What did you expect to happen?
App starts normally.

What was the actual behavior?
App fails to start with the following:

```
   2018-01-25T14:15:30.61-0700 [APP/PROC/WEB/0] ERR Error:
   2018-01-25T14:15:30.61-0700 [APP/PROC/WEB/0] ERR   An assembly specified in the application dependencies manifest (testapp.deps.json) was not found:
   2018-01-25T14:15:30.61-0700 [APP/PROC/WEB/0] ERR     package: 'Microsoft.AspNetCore.Mvc.Abstractions', version: '2.0.2'
   2018-01-25T14:15:30.61-0700 [APP/PROC/WEB/0] ERR     path: 'lib/netstandard2.0/Microsoft.AspNetCore.Mvc.Abstractions.dll'
   2018-01-25T14:15:30.61-0700 [APP/PROC/WEB/0] ERR   This assembly was expected to be in the local runtime store as the application was published using the following target manifest files:
   2018-01-25T14:15:30.61-0700 [APP/PROC/WEB/0] ERR     aspnetcore-store-2.0.5.xml
   2018-01-25T14:15:30.63-0700 [APP/PROC/WEB/0] OUT Exit status 140
   2018-01-25T14:15:30.63-0700 [CELL/SSHD/0] OUT Exit status 0
   2018-01-25T14:15:30.65-0700 [API/18] OUT Process has crashed with type: "web"
   2018-01-25T14:15:30.66-0700 [API/18] OUT App instance exited with guid 6af71e09-c141-4371-964d-9b2f24fd5d64 payload: {"instance"=>"2a661b4f-2f1b-4cb0-725d-7c5c", "index"=>0, "reason"=>"CRASHED", "exit_description"=>"APP/PROC/WEB: Exited with status 140", "crash_count"=>1, "crash_timestamp"=>1516914930632167538, "version"=>"14c45c5c-a6c5-40a1-8aca-e065c8a4a745"}
   2018-01-25T14:15:30.68-0700 [CELL/0] OUT Stopping instance 2a661b4f-2f1b-4cb0-725d-7c5c
   2018-01-25T14:15:30.68-0700 [CELL/0] OUT Destroying container
   2018-01-25T14:15:30.84-0700 [CELL/0] OUT Creating container
   2018-01-25T14:15:31.31-0700 [CELL/0] OUT Successfully created container
   2018-01-25T14:15:31.83-0700 [CELL/0] OUT Successfully destroyed container
   2018-01-25T14:15:40.42-0700 [CELL/0] OUT Starting health monitoring of container
   2018-01-25T14:15:40.62-0700 [APP/PROC/WEB/0] ERR Error:
   2018-01-25T14:15:40.62-0700 [APP/PROC/WEB/0] ERR   An assembly specified in the application dependencies manifest (testapp.deps.json) was not found:
   2018-01-25T14:15:40.62-0700 [APP/PROC/WEB/0] ERR     package: 'Microsoft.AspNetCore.Mvc.Abstractions', version: '2.0.2'
   2018-01-25T14:15:40.62-0700 [APP/PROC/WEB/0] ERR     path: 'lib/netstandard2.0/Microsoft.AspNetCore.Mvc.Abstractions.dll'
   2018-01-25T14:15:40.62-0700 [APP/PROC/WEB/0] ERR   This assembly was expected to be in the local runtime store as the application was published using the following target manifest files:
   2018-01-25T14:15:40.62-0700 [APP/PROC/WEB/0] ERR     aspnetcore-store-2.0.5.xml
   2018-01-25T14:15:40.63-0700 [APP/PROC/WEB/0] OUT Exit status 140
   2018-01-25T14:15:40.64-0700 [CELL/SSHD/0] OUT Exit status 0
   2018-01-25T14:15:40.77-0700 [API/0] OUT Process has crashed with type: "web"
   2018-01-25T14:15:40.78-0700 [API/0] OUT App instance exited with guid 6af71e09-c141-4371-964d-9b2f24fd5d64 payload: {"instance"=>"b18e5b60-ff6a-4754-7613-0cce", "index"=>0, "reason"=>"CRASHED", "exit_description"=>"APP/PROC/WEB: Exited with status 140", "crash_count"=>2, "crash_timestamp"=>1516914940750871945, "version"=>"14c45c5c-a6c5-40a1-8aca-e065c8a4a745"}
   2018-01-25T14:15:40.78-0700 [CELL/0] OUT Stopping instance b18e5b60-ff6a-4754-7613-0cce
   2018-01-25T14:15:40.78-0700 [CELL/0] OUT Destroying container
   2018-01-25T14:15:40.96-0700 [CELL/0] OUT Creating container
   2018-01-25T14:15:41.49-0700 [CELL/0] OUT Successfully destroyed container
   2018-01-25T14:15:41.54-0700 [CELL/0] OUT Successfully created container
   2018-01-25T14:15:57.34-0700 [CELL/0] OUT Starting health monitoring of container
   2018-01-25T14:15:57.63-0700 [APP/PROC/WEB/0] ERR Error:
   2018-01-25T14:15:57.63-0700 [APP/PROC/WEB/0] ERR   An assembly specified in the application dependencies manifest (testapp.deps.json) was not found:
   2018-01-25T14:15:57.63-0700 [APP/PROC/WEB/0] ERR     package: 'Microsoft.AspNetCore.Mvc.Abstractions', version: '2.0.2'
   2018-01-25T14:15:57.63-0700 [APP/PROC/WEB/0] ERR     path: 'lib/netstandard2.0/Microsoft.AspNetCore.Mvc.Abstractions.dll'
   2018-01-25T14:15:57.63-0700 [APP/PROC/WEB/0] ERR   This assembly was expected to be in the local runtime store as the application was published using the following target manifest files:
   2018-01-25T14:15:57.63-0700 [APP/PROC/WEB/0] ERR     aspnetcore-store-2.0.5.xml
```

Please confirm where necessary:
* [x] I have included a log output
* [x] My log includes an error message
* [x] I have included steps for reproduction

Additional information:

1. If the `cf push` path points to the source rather than the published artifacts, the app deploys and starts correctly.

   ```
   cf push testapp --random-route -p .
   ```

1. If the `Microsoft.AspNetCore.All` package version is bumped down from `2.0.5` to `2.0.3`, the app deploys and starts correctly. (I'm assuming the buildpack pulls down the necessary dependencies when it builds the project.)

So I think it's a matter of not having the latest (2.0.5) assemblies available in the runtime package store.

**Comment by sartan** [comment]
Including `testapp.deps.json` if that's useful.
[testapp.deps.json.txt](https://github.com/cloudfoundry/dotnet-core-buildpack/files/1665745/testapp.deps.json.txt)

**Comment by dhruvanmurthy** [comment]
Ran into same issue:
`
Error:
  An assembly specified in the application dependencies manifest (mywebapp.deps.json) was not found:
    package: 'Microsoft.AspNetCore.Mvc.Abstractions', version: '2.0.2'
    path: 'lib/netstandard2.0/Microsoft.AspNetCore.Mvc.Abstractions.dll'
  This assembly was expected to be in the local runtime store as the application was published using the following target manifest files:
    aspnetcore-store-2.0.5.xml
`

**Comment by ateregulov** [comment]
Same problems

**Comment by sclevine** [comment]
This was addressed in 2.1.1. See the details here:
https://github.com/cloudfoundry/dotnet-core-buildpack/issues/184#issuecomment-401079194
